homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -355,14 +355,17 @@ def __init__(
         self.config_entry = config_entry
         self.implementation = implementation
 
+    @property
+    def token(self) -> dict:
+        """Return the current token."""
+        return cast(dict, self.config_entry.data["token"])
+
     async def async_ensure_token_valid(self) -> None:
         """Ensure that the current token is valid."""
-        token = self.config_entry.data["token"]
-
-        if token["expires_at"] > time.time():
+        if self.token["expires_at"] > time.time():
             return
 
-        new_token = await self.implementation.async_refresh_token(token)
+        new_token = await self.implementation.async_refresh_token(self.token)
```

Comment:
Why implement your own check on the token expire date? Request oauth was already doing that. I don't see the difference.

Reply:
This is code for the generic OAuth2 session inside Home Assistant. This is not specific code for the Somfy integration. So if called in a different contact, no check has been done yet.